app/controllers/schemes_controller.rb

```diff
@@ -98,7 +85,22 @@ def check_answers
     render "schemes/check_answers"
   end
 
-private
+  private
+
+  def derive_next_path page
```

Comment:
Personal style thing/just a nit but I think where possible naming methods with nouns after the thing the return is more idiomatic than naming them with a verb after the thing they do (i.e. `next_page_path` vs `derive_next_page`). (Unless you're deliberately doing something procedural I suppose).